reference/auth.v2.yaml

```diff
@@ -0,0 +1,270 @@
+              Token Exchange Request:
+                value:
+                  client_id: cgm-monitor
+                  client_secret: c50e6502-131c-47f0-b439-a43acb3b83d0
+                  grant_type: 'urn:ietf:params:oauth:grant-type:token-exchange'
+                  subject_issuer: acme-clinic
+                  subject_token: eyJhbGciOiJ...
+                  subject_token_type: 'urn:ietf:params:oauth:token-type:access_token'
+                  requested_token_type: 'urn:ietf:params:oauth:token-type:refresh_token'
+        description: ''
+  '/realms/{realm}/protocol/openid-connect/auth':
+    parameters:
+      - $ref: '#/components/parameters/realm'
+    get:
+      summary: Authorize
+      tags: []
+      responses: {}
+      operationId: Authorize
+      description: |
+        The starting point for browser-based OpenID Connect flows. This endpoint authenticates the user and returns an authorization grant or an access token to callback endpoint at the specified `redirect_uri`.
+      parameters:
+        - schema:
+            type: string
+          in: query
+          name: client_id
+          description: Client Identifier valid at the Authorization Server
+          required: true
+        - schema:
+            type: array
+            enum:
+              - openid
+              - email
+          in: query
+          name: scope
+          required: true
+          style: spaceDelimited
+          description: OpenID Connect requests MUST contain the `openid` scope value
+        - schema:
+            type: string
+            enum:
+              - code
+          in: query
+          name: respone_type
+          required: true
+          description: 'Response Type value that determines the authorization processing flow to be used, including what parameters are returned from the endpoints used. When using the Authorization Code Flow, this value is `code`.'
+        - schema:
+            type: string
+          in: query
+          name: redirect_uri
+          required: true
+          description: ' Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider'
+        - schema:
+            type: string
+          in: query
+          name: login_hint
+          description: Hint to the Authorization Server about the login identifier the End-User might use to log in
+        - schema:
+            type: string
+          in: query
+          name: kc_idp_hint
+          description: OIDC applications can bypass the login page by hinting at the identity provider they want to use
+        - schema:
+            type: string
+            enum:
+              - none
+              - login
+          in: query
+          name: prompt
+          description: Specifies whether the Authorization Server prompts the End-User for reauthentication and consent
+components:
+  schemas:
+    TokenRequest:
```

Comment:
Future improvement: we may want to define base type + different variants of this schema for each `grant_type` since different fields are required depending on the grant type.

Reply:
`oneOf` has poor support especially in code generation tools so it's more problematic than maintaining a flat schema. I prefer having specific examples and descriptions for the different flows / sets of supported parameters, otherwise we'd end up with duplicated values across different polymorphic types and/or deeply nested types (which also don't have good support in some tools).

Reply:
👍🏻 yep, the base `data` model is a true torture test of all that. Even OAS3 rendering tools struggle with that, and I can only imagine how badly codegen tools choke on it.